### PR TITLE
Merge `main` branch into `release/2.6`

### PR DESCRIPTION
**Pull request by jld-dehn-hpe**
## Summary and Scope

We want to delete the `main` branch in this repo and make `release/2.6` the default branch now that the SAT product and documentation have moved into CSM (with the CSM 1.6 release). Before we do this, we need to ensure no work that should be included from `main` is missing in `release/2.6`.

With merging `main` into `release/2.6`, five YAML files were updated. The only merge conflict was a difference in lines 3-7 of the `sat_2.6_release_notes.md` file. The version numbers in `release/2.6` had been updated more recently than in `main`. This was probably as a result of PR #136 and PR #143 that were made directly to the release branch since we knew there would be no independent releases of SAT going forward.

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [x] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: .github/workflows/backport.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#
+# MIT License
+#
+# (C) Copyright 2023 Hewlett Packard Enterprise Development LP
```
Comment (jld-dehn-hpe) [inline .github/workflows/backport.yml]: We can update this license to `2023-2024` if you want.

**Comment by jld-dehn-hpe** [comment]
I think the license check problem will be fixed if I merge this PR after PR #144.

File: .github/workflows/backport.yml (hunk 1 above)
Comment (annapoorna-s-alt) [reply .github/workflows/backport.yml]: Since for every other files you updated, I think its good to update here as well. No harm!

**Review by annapoorna-s-alt — APPROVED** [review]

File: .github/workflows/backport.yml (hunk 1 above)
Comment (haasken-hpe) [reply .github/workflows/backport.yml]: Yep, I would go ahead and update this year to make license-check happy.

**Review by haasken-hpe — APPROVED** [review]

File: .github/workflows/backport.yml (hunk 1 above)
Comment (jld-dehn-hpe) [reply .github/workflows/backport.yml]: I'll update in the next commit.

**Pull request merged**